frontend/src/services/bookingApi.ts

```diff
@@ -0,0 +1,185 @@
+  } catch (error: any) {
+    return { activeBookings: null, error: `Error fetching data: ${error.message}` };
+  }
+};
+
+// Bike Status Counter
+export const bikeStatusCounterFetchApi = async () => {
+  let bikeCountFree: number = 0;
+  let bikeCountBooked: number = 0;
+  let bikeCountInUse: number = 0;
+  let bikeCountDisabled: number = 0;
+  try {
+    const response = await fetch(`${api.baseUrl + bikeStatusCounterUrl}`);
+    if (!response.ok) {
+      if (response.status === 404) {
+        throw new Error('Data not found');
+      } else if (response.status === 500) {
+        throw new Error('Server error');
+      } else {
+        throw new Error(`Error: ${response.status} - ${response.statusText}`);
+      }
+    }
+
+    console.log(`Status: ${response.status}`)
+
+    return {
+      bikeCountFree,
+      bikeCountBooked,
+      bikeCountInUse,
+      bikeCountDisabled,
+      error: null
+    };
+  } catch (error: any) {
+    return {
+      bikeCountFree: number,
+      bikeCountBooked: number,
+      bikeCountInUse: number,
+      bikeCountDisabled: number,
+      error: `Error fetching data: ${error.message}`
+    };
+  }
+}
+
+//Create Single Booking
+export const createSingleBookingFetchApi = async (bookingData: Booking) => {
+  try {
+    const response = await fetch(api.baseUrl + createSingleBookingUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        userName: bookingData.bookingUserData.firstName + " " + bookingData.bookingUserData.lastName,
+        room: bookingData.bookingUserData.roomNumber,
+        bikeSize: bookingData.bookingBikeSize,
+      }),
+    });
+    console.log("User: " + bookingData.bookingUserData.firstName + " " + bookingData.bookingUserData.lastName)
+    if (!response.ok) {
+      if (response.status === 404) {
+        throw new Error('Data not found');
+      } else if (response.status === 500) {
+        throw new Error('Server error');
+      } else {
+        const errorData = await response.json();
+        throw new Error(errorData.error);
+      }
+    }
+
+    const data = await response.json();
+    return { booking: data.booking, error: null }; // Assuming the server returns the booking information
+  } catch (error: any) {
+    console.error('Error creating single booking:', error.message);
+    return {
+      data: null, error: `Error fetching data: ${error.message}`
+    }
+  }
+}
+
+// Approve a booking
+export const approveBookingFetchApi = async (bookingId: number) => {
+  try {
+    console.log(bookingId)
+    const response = await fetch(api.baseUrl + approveBookingUrl + bookingId, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    if (!response.ok) {
+      if (response.status === 404) {
+        throw new Error('Data not found');
+      } else if (response.status === 500) {
+        throw new Error('Server error');
+      } else {
```

Comment:
besides 404, 500, the API can return a 401 as described in IB-48. Could you handle this code as well? 

Reply:
Thank you for the advice. In IB-50 I changed the code to handle all returns at the same time. It's already in the main branch.